### ISSUE TITLE
Add testing support

### ISSUE THREAD

**Issue by tomsarduy** [ndelvalle/generator-api#29]
I really like the concept behind this starter and I will be happy to contribute at some point. I think one of the few things missing is the possibility to the user to add testing or not.

**Comment by gillchristian** [comment]
Awesome!

Yes I think it would be great to include tests.

**Comment by ndelvalle** [comment]
Testing is something I've being wanting to implement to this generator for long. I've being using [jest](https://facebook.github.io/jest/) lately and I really liked it. It works fine on node projects as well (Is not just for react). What do you guys think? I know is not wildly used, but maybe ppl will appreciate the simplicity and luck of configuration.
And I would really appreciate some help 🙂

**Comment by tomsarduy** [comment]
I agree 100%, I love Jest. I've been using it for a while in my react projects, and I think definitely can work with node/express as well. They have been doing a very good work in the last few releases, and it's based on Jasmine, so some people may be familiar with it. On the other side, I'm aware Mocha/Chai is what most of the people use.

**Comment by pthieu** [comment]
Any updates on this?

**Comment by Sigkill9** [comment]
Let me first say thanks! I really like this generator. I recommend and use this generator as a starting point all the time. Adding testing would be amazing. Sinon, mocha, chai or whatever. This is the one missing piece. Any plans to implement something anytime soon? Thanks.

**Comment by ndelvalle** [comment]
Thanks for the support @Sigkill9, I'm kind of busy, but I think this is really important. So I'm thinking of starting a branch and working on this with @gillchristian and anyone that wants to join next weekend.
